rfc/rfc-99/rfc-99.md

```diff
@@ -0,0 +1,219 @@
+
+| Logical Type | Description | Parameters |
+| :---- | :---- | :---- |
+| BOOLEAN | A logical boolean value (true/false). | None |
+| TINYINT | An 8-bit signed integer. | None |
+| UINT8 | An 8-bit **unsigned** integer. | None |
+| SMALLINT | A 16-bit signed integer. | None |
+| UINT16 | A 16-bit **unsigned** integer. | None |
+| INTEGER | A 32-bit signed integer. | None |
+| UINT32 | A 32-bit **unsigned** integer. | None |
+| BIGINT | A 64-bit signed integer. | None |
+| UINT64 | A 64-bit **unsigned** integer. | None |
+| FLOAT16 | A 16-bit half-precision floating-point number. | None |
+| FLOAT | A 32-bit single-precision floating-point number. | None |
+| DOUBLE | A 64-bit double-precision floating-point number. | None |
+| DECIMAL(p, s) | An exact numeric with specified precision/scale. | p, s |
+| STRING | A variable-length UTF-8 character string, limited to 2GB per value. | None |
+| LARGE\_STRING | A variable-length UTF-8 character string for values exceeding 2GB. | None |
+| BINARY | A variable-length sequence of bytes, limited to 2GB per value. | None |
+| LARGE\_BINARY | A variable-length sequence of bytes for values exceeding 2GB. | None |
+| FIXED(n) | A fixed-length sequence of n bytes. | n |
+| UUID | A 128-bit universally unique identifier. | None |
+
+#### **3.2. Temporal Types**
+
+These types handle date and time representations with high precision and timezone awareness.
+
+| Logical Type | Description | Parameters |
+| :---- | :---- | :---- |
+| DATE | A calendar date (year, month, day). | None |
+| DATE64 | A calendar date stored as milliseconds. | None |
+| TIME(precision) | A time of day without a timezone. | s, ms, us, ns |
+| TIMESTAMP(precision) | An instant in time without a timezone. | us or ns |
+| TIMESTAMPTZ(precision) | An instant in time with a timezone, normalized and stored as UTC. | us or ns |
+| DURATION(unit) | An exact physical time duration, independent of calendars. | s, ms, us, ns |
+| INTERVAL | Represents a duration of time (e.g., months, days, milliseconds). | None |
+
+#### **3.3. Composite Types**
+
+These types allow for the creation of complex, nested data structures.
+
+| Logical Type | Description | Parameters |
+| :---- | :---- | :---- |
+| STRUCT\<name: type, ...\> | An ordered collection of named fields. | Field list |
+| LIST\<element\_type\> | An ordered list of elements of the same type. | Element type |
+| MAP\<key\_type, value\_type\> | A collection of key-value pairs. Keys must be unique. | Key, Value types |
+| UNION\<type1, type2, ...\> | A value that can be one of several specified types. | Type list |
```

Comment:
is UNION in SQL standard?

Reply:
Don't think so. Kindly look at the above comment. I am not taking this as an objective.